Doc/library/asyncio-task.rst

```diff
@@ -861,19 +861,46 @@ Waiting Primitives
 
 .. function:: as_completed(aws, *, timeout=None)
 
-   Run :ref:`awaitable objects <asyncio-awaitables>` in the *aws*
-   iterable concurrently.  Return an iterator of coroutines.
-   Each coroutine returned can be awaited to get the earliest next
-   result from the iterable of the remaining awaitables.
+   Run :ref:`awaitable objects <asyncio-awaitables>` in the *aws* iterable
+   concurrently. The returned object can be iterated to obtain earliest next
+   results of the awaitables as they finish.
+
+   The object returned by ``as_completed()`` can be iterated as an
+   :term:`asynchronous iterator` or a plain :term:`iterator`. When asynchronous
+   iteration is used, the originally-supplied awaitables are yielded if they
+   are tasks or futures. This makes it easy to correlate previously-scheduled
+   tasks with their results. Example::
+
+       ipv4_connect = create_task(
+           open_connection("127.0.0.1", 80)
+       )
+       ipv6_connect = create_task(
+           open_connection("::1", 80)
+       )
+       tasks = [ipv4_connect, ipv6_connect]
```

Comment:
This could be written more compactly:
```suggestion
       tasks = [
           create_task(open_connection("127.0.0.1", 80)),
           create_task(open_connection("::1", 80)),
       ]
```

Reply:
We need the `ipv6_connect` variable for use in the identity check below.